specs/manage_tokens_list/01_main.rst

```diff
@@ -0,0 +1,137 @@
+==========================================================
+Manage Tokens - Selection List
+==========================================================
+
+=======  ====================  =======    ==========
+ epic          title           author     created
+=======  ====================  =======    ==========
+58_      Token Selection List  rmeissner  23.01.2019
+=======  ====================  =======    ==========
+
+.. _58: https://github.com/gnosis/safe/issues/58
+
+.. _Main:
+
+#. `Main`_
+#. `User Interface`_
+#. `Technical Details`_
+
+.. sectnum::
+.. contents:: Table of Contents
+    :local:
+    :depth: 2
+
+Problem Definition
+-------------------------------
+
+- User MUST be able to see all token supported by the Safe app
+- User MUST be able to enable and disable tokens that should be displayed in the asset overview 
+
+
+Outputs
+------------
+
+- Selected tokens for asset overview
+
+Use Case Scenarios
+-----------------------
+
+Load token list
+~~~~~~~~~~~~~~~~~~~~
+
+Steps
++++++
+1. The content (token list) should start loading automatically
+2. Once loading is done the tokens should be displayed
+
+Postconditions
+++++++++++++++
+- All available tokens should be displayed after the selected tokens.
+
+  * It should be possible to display an unlimited number of tokens in the list (paginated list)
+  * The next page of tokens should be loaded automatically when scrolling close to the end of the list
+  * It should be indicated that data is loading
+  
+- For each token we display the details
+
+  * Icon
+  * Name
+  * Symbol
+  * Enabled or not
+
+- Loaded list: https://zpl.io/aRqlmBe
+
+
+Enable token from list
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+inherits from `Load token list`_
+
+Steps
++++++++
+3. Select token after list has been loaded
+
+Postconditions
+++++++++++++++
+* Token should be displayed in asset overview
+
+
+Disable token from list
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+inherits from `Load token list`_
+
+Preconditions
++++++++++++++
+
+- A token is already enabled (and displayed in the asset overview)
+
+Steps
++++++++
+3. Select token after list that is enable and disable it
+
+Postconditions
+++++++++++++++
+- Token should NOT be displayed in asset overview
+
+
+
+Could not load initial token list
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+inherits from `Load token list`_
+
+Steps
++++++++
+2. An non-blocking  error should be displayed if the list could not be loaded
```

Comment:
Currently, on iOS, we have a "retry with increasing delay" logic and do not show an error. The retrieved results and store them in db. Token list is updated on each sync. We do the tokens list sync on app start and in some other places.
We also have a merging tokens logic that takes into account "whitelisted" and "blacklisted" tokens.